File: conteudo-curso/aulas-visual-basic/December/MeuProjeto - Teilor_02-12/MeuProjeto - Teilor/MeuProjeto/Produto.vb
Imports MeuProjeto.BancoDeDados

Public Class Produto
    ' Atributos
    Private _id As Integer
    Private _descricao As String
    Private _preco As Double
    Private _estoque As Integer
    Private banco As New BancoDeDados

    Public Property Id As Integer
        Get
            Return _id
        End Get
        Set(value As Integer)
            _id = value
        End Set
    End Property

    Public Property Descricao As String
        Get
            Return _descricao
        End Get
        Set(value As String)
            _descricao = value
        End Set
    End Property

    Public Property Preco As Double
        Get
            Return _preco
        End Get
        Set(value As Double)
            _preco = value
        End Set
    End Property

    Public Property Estoque As Integer
        Get
            Return _estoque
        End Get
        Set(value As Integer)
            _estoque = value
        End Set
    End Property

    ' Método Inserir
    Public Sub Inserir()
        Try
            Me.banco.Conectar()
            Dim sql As String = "INSERT INTO 


s (descricao, preco, estoque) VALUES (?, ?, ?)"
            Me.banco.Executar(sql, Me.Descricao, Me.Preco, Me.Estoque)
            MsgBox("Produto cadastrado com sucesso!")
        Catch erro As Exception
            MsgBox("Houve um problema com o Banco de Dados: " & erro.Message)
        Finally
            Me.banco.Desconectar()
        End Try '
    End Sub

    ' Método Atualizar
    Public Sub Atualizar()
        Try
            Me.banco.Conectar()
            Dim sql As String = "UPDATE produtos SET descricao = ?, preco = ?, estoque = ? WHERE id_produto = ?"
            Me.banco.Executar(sql, Me.Descricao, Me.Preco, Me.Estoque, Me.Id)
            MsgBox("Produto alterado com sucesso!")
        Catch erro As Exception
            MsgBox("Houve um problema com o Banco de Dados: " & erro.Message)
        Finally
            Me.banco.Desconectar()
        End Try
    End Sub

    ' Excluir
    Public Sub excluir()
        Try
            Me.banco.Conectar()
            Dim sql As String = "DELETE FROM produtos WHERE id_produto = ?"
            Me.banco.Executar(sql, Me.Id)
            MsgBox("Produto removido com sucesso!")
        Catch erro As Exception
            MsgBox("Houve um problema com o Banco de Dados: " & erro.Message)
        Finally
            Me.banco.Desconectar()
        End Try
    End Sub


    ' Selecionar por ID (Retonar um VETOR de dados DataRow)
    Public Function SelecionarPorId() As DataRow
        Try
            Me.banco.Conectar()
            Dim sql As String = "SELECT * FROM produtos WHERE id_produto = ?"
            Dim dados As DataTable = Me.banco.Consultar(sql, Me.Id)
            Return dados(0)     ' Retornando a primeira e única linha da consulta (DataTable)
        Catch erro As Exception
            MsgBox("Houve um problema com o Banco de Dados: " & erro.Message)
            Return Nothing
        Finally
            Me.banco.Desconectar()
        End Try
    End Function

    ' Selecionar Todos
    Public Function SelecionarTodos() As DataTable
        Try
            Me.banco.Conectar()
            Dim sql As String = "SELECT * FROM produtos"
            Return Me.banco.Consultar(sql)
        Catch erro As Exception
            MsgBox("Houve um problema com o Banco de Dados: " & erro.Message)
            Return Nothing
        Finally
            Me.banco.Desconectar()
        End Try
    End Function
End Class

File: conteudo-curso/aulas-desenv-web/November/Aula 2711/10 - sistema + poo/src/class/Produto.php
<?php

require_once 'BancoDeDados.php';
// use BancoDeDados;

class Produto {
    // Atributos
    public string $id;
    public string $descricao;
    public string $preco;
    public string $estoque;
    public string $imagem;
    private BancoDeDados $banco;

    // Construtor
    public function __construct()
    {
        $this->banco = new BancoDeDados;
    }

    // Método Inserir
    public function inserir()
    {
        try {
            $sql = 'INSERT INTO produtos (
                        descricao,
                        preco,
                        estoque,
                        imagem
                    ) VALUES (?,?,?,?)';
            $parametros = [
                $this->descricao,
                $this->preco,
                $this->estoque,
                $this->imagem,
            ];
            $this->banco->executarComando($sql, $parametros);
        } catch (PDOException $erro) {
            throw new PDOException;
        }
    }

    // Método Atualizar
    public function atualizar()
    {
        try {
            $sql = 'UPDATE produtos SET
                    descricao = ?,
                    preco = ?,
                    estoque = ?
                    WHERE id_produto = ?';
            $parametros = [
                $this->descricao,
                $this->preco,
                $this->estoque,
                $this->id,
            ];
            $this->banco->executarComando($sql, $parametros);
        } catch (PDOException $erro) {
            throw new PDOException;
        }
    }

    // Método Deletar
    public function deletar()
    {
        try {
            $sql = 'DELETE FROM produtos
                    WHERE id_produto = ?';
            $parametros = [
                $this->id,
            ];
            $this->banco->executarComando($sql, $parametros);
        } catch (PDOException $erro) {
            throw new PDOException;
        }
    }

    // Método Selecionar Por ID
    public function selecionarPorId()
    {
        try {
            $sql = 'SELECT * FROM produtos
                    WHERE id_produto = ?';
            $parametros = [
                $this->id,
            ];
            return $this->banco->consultar($sql, $parametros);
        } catch (PDOException $erro) {
            throw new PDOException;
        }
    }

    // Método Selecionar Por ID
    public function selecionarTodos()
    {
        try {
            $sql = 'SELECT * FROM produtos';
            return $this->banco->consultar($sql, [], true);
        } catch (PDOException $erro) {
            throw new PDOException;
        }
    }
}